birdset/datamodule/beans_datamodule.py

```diff
@@ -13,18 +16,78 @@ def __init__(
             dataset: DatasetConfig = DatasetConfig(),
             loaders: LoadersConfig = LoadersConfig(),
             transforms: BirdSetTransformsWrapper = BirdSetTransformsWrapper(),
+            mapper: None = None,
+            k_samples: int = 0
     ):
         super().__init__(
             dataset=dataset,
             loaders=loaders,
             transforms=transforms
         )
+        self.k_samples = k_samples
+        self.id_to_label = defaultdict(str)
 
     def _preprocess_data(self, dataset):
         """
-        Preprocess the data. If multilabel is the task we will one hot encode.
+        Preprocess the data. If multilabel is the task we will one hot encode. Use k_samples > 0 if you want control over amount of samples per class. The rest is used for validation and testing.
         """
 
+        if self.k_samples > 0:
```

Comment:
Put this into a separate function to increase readability and to also use in other Datamodules.